great_expectations/data_context/store/expectations_store.py

```diff
@@ -197,40 +222,34 @@ def _add(self, key, value, **kwargs):
             if self.cloud_mode:
                 # cloud backend has added IDs, so we update our local state to be in sync
                 result = cast(GXCloudResourceRef, result)
-
-                suite_kwargs = result.response["data"]["attributes"]["suite"]
-
-                # Temporary fork to account for pre-V1 configs
-                if "ge_cloud_id" in suite_kwargs and "id" not in suite_kwargs:
-                    suite_kwargs["id"] = suite_kwargs.pop("ge_cloud_id")
-
+                suite_kwargs = self.deserialize(
+                    self.gx_cloud_response_json_to_object_dict(result.response)
+                )
                 cloud_suite = ExpectationSuite(**suite_kwargs)
                 value = self._add_cloud_ids_to_local_suite_and_expectations(
                     local_suite=value,
                     cloud_suite=cloud_suite,
                 )
             return result
-        except gx_exceptions.StoreBackendError:
+        except gx_exceptions.StoreBackendError as exc:
             raise gx_exceptions.ExpectationSuiteError(
                 f"An ExpectationSuite named {value.name} already exists."
-            )
+            ) from exc
 
     def _update(self, key, value, **kwargs):
         if not self.cloud_mode:
             # this logic should move to the store backend, but is implemented here for now
             value: ExpectationSuite = self._add_ids_on_update(value)
         try:
-            # todo: `update` should return the updated object
-            super()._update(key=key, value=value, **kwargs)
+            result = super()._update(key=key, value=value, **kwargs)
 
             if self.cloud_mode:
                 # cloud backend has added IDs, so we update our local state to be in sync
-                # todo: add back this logic when `update` returns the updated object
-                # result = cast(GXCloudResourceRef, result)
-                # cloud_suite = ExpectationSuite(
-                #     **result.response["data"]["attributes"]["suite"]
-                # )
-                _suite_identifier, cloud_suite = self._refresh_suite(value)
+                result = cast(GXCloudResourceRef, result)
+                suite_kwargs = self.deserialize(
```

Comment:
this coverage check is incorrect; see `test_expectations_store.test_update_cloud_suite`.